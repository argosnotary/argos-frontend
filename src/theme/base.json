{
  "fontFamily": "'Fira Sans'",
  "fontFile": "/fonts/fira-sans-v10-latin-regular",
  "button": {
    "bgColor": "#1779ba",
    "textColor": "#fff"
  },
  "cancelButton": {
    "bgColor": "transparent",
    "textColor": "#8D99AE",
    "borderColor": "#8D99AE",
    "hover": {
      "bgColor": "#8D99AE",
      "textColor": "#fff",
      "borderColor": "transparent"
    }
  },
  "transparentButton": {
    "default": {
      "transparentButtonBgColor": "#fff",
      "transparentButtonTextColor": "#1779ba",
      "transparentButtonBorderColor": "#1779ba"
    },
    "hover": {
      "transparentButtonBgColor": "#1779ba",
      "transparentButtonTextColor": "#fff",
      "transparentButtonBorderColor": "#1779ba"
    }
  },
  "alternateLoader": {
    "color": "#808080"
  },
  "loaderIcon": {
    "color": "#1779ba"
  },
  "loaderButton": {
    "loadingColor": "#fff"
  },
  "dataCheckbox": {
    "loaderColor": "#1779ba"
  },
  "searchInput": {
    "searchResultsBgColor": "#fff",
    "searchResultEntryHoverBg": "#e8feff",
    "selectedEntryBorderColor": "#e0e0e0",
    "searchResultsBorderColor": "#e0e0e0",
    "loaderIconColor": "#1779ba"
  },
  "collapsibleContainer": {
    "iconColor": "gray",
    "headerBgColor": "#e0e0e0",
    "bodyBorderColor": "#e0e0e0"
  },
  "selectList": {
    "defaultItemBgColor": "#fff",
    "selectedItemBgColor": "#e8feff",
    "iconColor": "#8d99ae"
  },
  "panel": {
    "icons": {
      "shrinkIcon": {
        "color": "#1779ba"
      },
      "enlargeIcon": {
        "color": "#1779ba"
      }
    }
  },
  "modal": {
    "bgColor": "#fff",
    "button": {
      "default": {
        "color": "#1779ba"
      },
      "hover": {
        "color": "#1779ba"
      }
    }
  },
  "passwordView": {
    "clipboardIcon": {
      "color": "#8D99AE",
      "hoverBgColor": "#e8feff"
    }
  },
  "copyInput": {
    "bgColor": "#e0e0e0"
  },
  "notificationsList": {
    "notificationBgColor": "#fff",
    "iconSize": 22,
    "iconColor": "#fff",
    "colors": {
      "WARNING": "orange",
      "ERROR": "red",
      "SUCCESS": "green"
    }
  },
  "layoutBuilder": {
    "segmentContainerBorderColor": "#1779ba",
    "segmentContainerTitleBgColor": "#1779ba",
    "iconColor": "#1779ba",
    "stepContainerBgColor": "#e0e0e0",
    "stepContainerTitleBorderColor": "#1779ba",
    "addStepButtonBorderColor": "#1779ba",
    "segmentTitleBgColor": "#fff",
    "segmentTitleHoverBorderColor": "#1779ba",
    "stepTitleBgColor": "#fff",
    "stepTitleHoverBorderColor": "#1779ba",
    "segmentsContainerTitleColor": "#fff"
  },
  "treeEditor": {
    "textColor": "#2e2e2e",
    "bgColor": "#f1f1f1",
    "borderColor": "#e0e0e0",
    "loaders": {
      "onPageLoad": {
        "color": "#1779ba"
      },
      "onFetchChildren": {
        "color": "#1779ba"
      }
    },
    "iconColors": {
      "addRootNode": "#1779ba",
      "expandNode": "#1779ba",
      "label": "#8D99AE",
      "chain": "#8D99AE",
      "robot": "#8D99AE"
    },
    "treeHeadLabel": {
      "bgColor": "#e8feff"
    },
    "nodeContextMenuContainer": {
      "bgColor": "fff",
      "borderColor": "#e5e5e5"
    },
    "nodeContextMenuItem": {
      "iconColor": "#8D99AE",
      "bgColor": "#fff",
      "hover": {
        "bgColor": "#e8feff"
      }
    },
    "breadCrumb": {
      "textColor": "#1779ba"
    },
    "lastBreadCrumb": {
      "textColor": "#000"
    },
    "nodeContextMenuItemSeparator": {
      "borderColor": "#e5e5e5"
    },
    "nodeContextMenuIconSeparator": {
      "borderColor": "#8D99AE"
    }
  },
  "dashboardNavBar": {
    "link": {
      "textColor": "#2e2e2e",
      "hover": {
        "bgColor": "#e8feff"
      }
    }
  },
  "connectionError": {
    "textColor": "#fff",
    "bgColor": "red"
  },
  "alerts": {
    "warning": {
      "color": "red"
    }
  },
  "PageHeader": {
    "bottomBorderColor": "#bbb"
  },
  "homePage": {
    "bgColor": "#eee",
    "navigation": {
      "bgColor": "#fff",
      "loginButton": {
        "default": {
          "loginButtonBgColor": "#fff",
          "loginButtonTextColor": "#1779ba",
          "loginButtonBorderColor": "#1779ba"
        },
        "hover": {
          "loginButtonBgColor": "#1779ba",
          "loginButtonTextColor": "#fff",
          "loginButtonBorderColor": "#1779ba"
        }
      }
    }
  },
  "dashboardPage": {
    "avatar": {
      "borderColor": "#1779ba"
    },
    "contextMenu": {
      "bgColor": "#fff",
      "borderColor": "#e5e5e5",
      "menuItemTextColor": "#2e2e2e",
      "menuItemBgColor": "#fff",
      "menuItemHoverBgColor": "#eee",
      "dividerBgColor": "#dfdfdf"
    }
  },
  "loginPage": {
    "bgColor": "#eee"
  },
  "userSettingsPage": {
    "bgColor": "#f1f1f1",
    "sidePanel": {
      "bgColor": "#e0e0e0",
      "sidePanelItem": {
        "bgColor": "#c7c7c7",
        "activeBgColor": "#f1f1f1",
        "highlightColor": "#1779ba",
        "fontColor": "#2E2E30"
      }
    }
  },
  "keyManagementPage": {
    "iconColor": "#1779ba",
    "passwordColor": "orange",
    "noActiveKeyWarning": {
      "textColor": "orange",
      "borderColor": "orange"
    }
  },
  "layoutPage": {
    "panel": {
      "bgColor": "#f1f1f1",
      "borderColor": "#e0e0e0",
      "contentSeparator": "#ccc"
    }
  }
}
