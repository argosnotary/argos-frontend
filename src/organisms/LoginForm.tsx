// Copyright 2020 Rabobank Nederland
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import styled from 'styled-components';
import Button from '../atoms/Button';
import FormInput from '../molecules/FormInput';

const LoginFormContainer = styled.div`
  background-color: #fff;
  margin: 10% auto;
  width: 20rem;
  box-shadow: 0 3px 20px 0px rgba(0, 0, 0, 0.1);
  padding: 1.75rem 2rem 2rem;
`;

const LoginFormHeader = styled.img`
  height: 4.25rem;
  margin: 2rem auto 4rem;
  display: flex;
  max-width: 100%;
`;

const LoginButton = styled(Button)`
  width: 100%;
  margin: 1rem 0;
  font-size: 1rem;
`;

const LoginForm: React.FC = () => (
  <LoginFormContainer>
    <LoginFormHeader src="images/logo.svg" />
    <form>
      <FormInput
        labelValue={'Username'}
        placeHolder={'Username'}
        formType={'text'}
      />
      <FormInput
        labelValue={'Password'}
        placeHolder={'Password'}
        formType={'password'}
      />
      <LoginButton>Login</LoginButton>
    </form>
  </LoginFormContainer>
);

export default LoginForm;
