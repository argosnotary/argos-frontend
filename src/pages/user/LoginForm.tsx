import React from "react";
import styled from "styled-components";
import { ConnectionErrorMessage } from "../../atoms/ConnectionError";
import { OAuthProvider } from "../../api/api";

import { AnchorButton } from "../../atoms/Button";

const AuthProviderImage = styled.img`
    width: 1.5rem;
`;

const AuthProviderLabel = styled.span``;

const AuthLabelAndImageSeparator = styled.span`
    border-right: 1px solid ${props => props.theme.loginForm.separatorColor};
    width: 1px;
    height: 1rem;
    margin: 0 1rem;
`;

const LoginButton = styled(AnchorButton)`
    border: 1px solid ${props => props.theme.loginForm.buttonBorderColor};
    color: ${props => props.theme.loginForm.buttonTextColor};

    &:hover {
        border-left: 4px solid ${props => props.theme.loginForm.buttonHoverHighlightColor};
    }
`;

const LoginFormHeader = styled.img`
    height: 4.25rem;
    margin: 2rem auto 4rem;
    display: flex;
    max-width: 100%;
`;

const StyledForm = styled.div`
    display: flex;
    flex-direction: column;
    background-color: ${props => props.theme.loginForm.bgColor};
    margin: 10% auto;
    width: 20rem;
    box-shadow: 0 3px 20px 0px rgba(0, 0, 0, 0.1);
    padding: 1.75rem 2rem 2rem;
`;

interface LoginProvidersProps {
    showError: boolean;
    providers: OAuthProvider[];
}

function LoginFormContainer(props: LoginProvidersProps): JSX.Element {
    return (
        <StyledForm>
            <LoginFormHeader src="images/logo.svg" />
            {props.showError ? <ConnectionErrorMessage>Connection error try again later</ConnectionErrorMessage> : null}
            {props.providers.map((provider, index) => {
                return (
                    <LoginButton
                        key={`provider${index}`}
                        href={`/api/oauth2/authorize/${provider.providerName}?redirect_uri=/authenticated`}>
                        {provider.iconUrl ? (
                            <>
                                <AuthProviderImage src={provider.iconUrl} />
                                <AuthLabelAndImageSeparator />
                            </>
                        ) : null}
                        <AuthProviderLabel>Login with {provider.displayName}</AuthProviderLabel>
                    </LoginButton>
                );
            })}
        </StyledForm>
    );
}

export default LoginFormContainer;
