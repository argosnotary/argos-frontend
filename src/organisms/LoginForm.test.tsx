/*
 * Copyright (C) 2019 - 2020 Rabobank Nederland
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { mount } from "enzyme";

import MockAdapter from "axios-mock-adapter";
import Axios from "axios";
import { ThemeProvider } from "styled-components";
import theme from "../theme/base.json";
import { act } from "react-dom/test-utils";
import LoginForm, { IProvider } from "./LoginForm";

const mock = new MockAdapter(Axios);

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: jest.fn()
  })
}));

it("renders correctly", async () => {
  const providers: Array<IProvider> = [
    {
      providerName: "provider1",
      displayName: "name1"
    },
    {
      providerName: "provider2",
      displayName: "name2",
      iconUrl: "http://localhost/icon.svg"
    }
  ];

  mock.onGet("/api/oauthprovider").reply(200, providers);

  const root = mount(
    <ThemeProvider theme={theme}>
      <LoginForm />
    </ThemeProvider>
  );

  await act(() =>
    new Promise(resolve => setImmediate(resolve)).then(() => {
      root.update();
      expect(root.find(LoginForm)).toMatchSnapshot();
    })
  );
});

it("renders correctly error", async () => {
  mock.onGet("/api/oauthprovider").reply(404);

  const root = mount(
    <ThemeProvider theme={theme}>
      <LoginForm />
    </ThemeProvider>
  );

  await act(() =>
    new Promise(resolve => setImmediate(resolve)).then(() => {
      root.update();
      expect(root.find(LoginForm)).toMatchSnapshot();
    })
  );
});
