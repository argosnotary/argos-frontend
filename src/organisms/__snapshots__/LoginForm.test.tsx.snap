// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<LoginForm>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": false,
            "lastClassName": "kZGGxV",
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
  background-color: ",
              [Function],
              ";
  margin: 10% auto;
  width: 20rem;
  box-shadow: 0 3px 20px 0px rgba(0, 0, 0, 0.1);
  padding: 1.75rem 2rem 2rem;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-EHOje kZGGxV"
      >
        <styled.img
          src="images/logo.svg"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bZQynM",
                  "isStatic": false,
                  "lastClassName": "XXprT",
                  "rules": Array [
                    "
  height: 4.25rem;
  margin: 2rem auto 4rem;
  display: flex;
  max-width: 100%;
",
                  ],
                },
                "displayName": "styled.img",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bZQynM",
                "target": "img",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            src="images/logo.svg"
          >
            <img
              className="sc-bZQynM XXprT"
              src="images/logo.svg"
            />
          </StyledComponent>
        </styled.img>
        <Styled(styled.a)
          href="/api/oauth2/authorize/provider1?redirect_uri=/authenticated"
          key="provider0"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gzVnrw",
                  "isStatic": false,
                  "lastClassName": "gODbxc",
                  "rules": Array [
                    "
  ",
                    "
  display: inline-block;
  vertical-align: middle;
  margin: 0 0 1rem 0;
  line-height: 1;
  padding: 0.8rem 1rem;
  border: 1px solid transparent;
  text-align: center;
  font-size: 0.9rem;
  font-family: inherit;
  background-color: ",
                    [Function],
                    ";
  color: ",
                    [Function],
                    ";
  outline: 0;
  transition: background-color 0.25s ease-out, color 0.25s ease-out;

  &:hover {
    background-color: ",
                    [Function],
                    ";
    cursor: pointer;
  }
",
                    "
",
                    "
  display: flex;
  justify-content: center;
  box-sizing: border-box;
  width: 100%;
  margin: 1rem 0;
  font-size: 1rem;
  text-decoration: none;
",
                  ],
                },
                "displayName": "Styled(styled.a)",
                "foldedComponentIds": Array [
                  "sc-bwzfXH",
                ],
                "render": [Function],
                "styledComponentId": "sc-gzVnrw",
                "target": "a",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            href="/api/oauth2/authorize/provider1?redirect_uri=/authenticated"
          >
            <a
              className="sc-bwzfXH sc-gzVnrw gODbxc"
              href="/api/oauth2/authorize/provider1?redirect_uri=/authenticated"
            >
              Login with 
              name1
               
            </a>
          </StyledComponent>
        </Styled(styled.a)>
        <Styled(styled.a)
          href="/api/oauth2/authorize/provider2?redirect_uri=/authenticated"
          key="provider1"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gzVnrw",
                  "isStatic": false,
                  "lastClassName": "gODbxc",
                  "rules": Array [
                    "
  ",
                    "
  display: inline-block;
  vertical-align: middle;
  margin: 0 0 1rem 0;
  line-height: 1;
  padding: 0.8rem 1rem;
  border: 1px solid transparent;
  text-align: center;
  font-size: 0.9rem;
  font-family: inherit;
  background-color: ",
                    [Function],
                    ";
  color: ",
                    [Function],
                    ";
  outline: 0;
  transition: background-color 0.25s ease-out, color 0.25s ease-out;

  &:hover {
    background-color: ",
                    [Function],
                    ";
    cursor: pointer;
  }
",
                    "
",
                    "
  display: flex;
  justify-content: center;
  box-sizing: border-box;
  width: 100%;
  margin: 1rem 0;
  font-size: 1rem;
  text-decoration: none;
",
                  ],
                },
                "displayName": "Styled(styled.a)",
                "foldedComponentIds": Array [
                  "sc-bwzfXH",
                ],
                "render": [Function],
                "styledComponentId": "sc-gzVnrw",
                "target": "a",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            href="/api/oauth2/authorize/provider2?redirect_uri=/authenticated"
          >
            <a
              className="sc-bwzfXH sc-gzVnrw gODbxc"
              href="/api/oauth2/authorize/provider2?redirect_uri=/authenticated"
            >
              Login with 
              name2
               
              <img
                src="http://localhost/icon.svg"
              />
            </a>
          </StyledComponent>
        </Styled(styled.a)>
      </div>
    </StyledComponent>
  </styled.div>
</LoginForm>
`;

exports[`renders correctly error 1`] = `
<LoginForm>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": false,
            "lastClassName": "kZGGxV",
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
  background-color: ",
              [Function],
              ";
  margin: 10% auto;
  width: 20rem;
  box-shadow: 0 3px 20px 0px rgba(0, 0, 0, 0.1);
  padding: 1.75rem 2rem 2rem;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-EHOje kZGGxV"
      >
        <styled.img
          src="images/logo.svg"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bZQynM",
                  "isStatic": false,
                  "lastClassName": "XXprT",
                  "rules": Array [
                    "
  height: 4.25rem;
  margin: 2rem auto 4rem;
  display: flex;
  max-width: 100%;
",
                  ],
                },
                "displayName": "styled.img",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bZQynM",
                "target": "img",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            src="images/logo.svg"
          >
            <img
              className="sc-bZQynM XXprT"
              src="images/logo.svg"
            />
          </StyledComponent>
        </styled.img>
        <styled.p>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "lastClassName": "fcaiKu",
                  "rules": Array [
                    "
  text-align: center;
  background-color: ",
                    [Function],
                    ";
  color: ",
                    [Function],
                    ";
  padding: 0.25rem 1rem;

  animation: ",
                    "8",
                    "s ",
                    Keyframes {
                      "id": "sc-keyframes-eXqJEs",
                      "inject": [Function],
                      "name": "eXqJEs",
                      "rules": Array [
                        "@-webkit-keyframes eXqJEs{0%{opacity:0;}10%{opacity:100;}90%{opacity:100;}100%{opacity:0;}}",
                        "@keyframes eXqJEs{0%{opacity:0;}10%{opacity:100;}90%{opacity:100;}100%{opacity:0;}}",
                      ],
                      "toString": [Function],
                    },
                    " ease-in-out forwards;
",
                  ],
                },
                "displayName": "styled.p",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <p
              className="sc-ifAKCX fcaiKu"
            >
              Connection error try again later
            </p>
          </StyledComponent>
        </styled.p>
      </div>
    </StyledComponent>
  </styled.div>
</LoginForm>
`;
